### PR TITLE
Add files to test client

### PR DESCRIPTION
**Pull request by CoalaJoe**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tickets       | #3299
| License       | MIT
| Doc PR        | api-platform/docs#...

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/Test/Client.php [hunk 1]
```diff
@@ -84,7 +84,7 @@ public function setDefaultOptions(array $defaultOptions): void
      *
      * @return Response
      */
-    public function request(string $method, string $url, array $options = []): ResponseInterface
+    public function request(string $method, string $url, array $options = [], array $files = []): ResponseInterface
```
Comment (soyuka) [inline src/Bridge/Symfony/Bundle/Test/Client.php:87]: We can't change the method sign here as it should respect this interface: https://github.com/symfony/contracts/blob/master/HttpClient/HttpClientInterface.php#L84. Files aren't in the option list, we could maybe support this via the `extra` key?

Comment (CoalaJoe) [reply src/Bridge/Symfony/Bundle/Test/Client.php]: Updated to your suggestion

File: src/Bridge/Symfony/Bundle/Test/Client.php [hunk 2]
```diff
@@ -121,7 +122,8 @@ public function request(string $method, string $url, array $options = []): Respo
             'url' => $resolvedUrl,
             'primary_port' => 'http:' === $url['scheme'] ? 80 : 443,
         ];
-        $this->kernelBrowser->request($method, $resolvedUrl, [], [], $server, $options['body'] ?? null);
+        $extra = $options['extra'] ?? [];
+        $this->kernelBrowser->request($method, $resolvedUrl, [], $extra['files'] ?? [], $server, $options['body'] ?? null);
```
Comment (soyuka) [inline src/Bridge/Symfony/Bundle/Test/Client.php:126]: you can use `$options['extra']['files'] ?? []` directly here

Comment (CoalaJoe) [reply src/Bridge/Symfony/Bundle/Test/Client.php]: I've updated it based on your suggestion

**Comment by soyuka** [comment]
Thanks @CoalaJoe we could maybe add some documentation to that in api-platform/docs!

**Comment by dunglas** [comment]
As reported on Slack, this should be backported in 2.5 as it's a bug fix.

**Pull request merged**